elastic/elastic-agent
# Avoid exiting with os.Exit

## What does this PR do?

This PR avoids exiting with os.Exit to enable defer functions cleanup properly.

## Why is it important?

Mainly needed for reporting `Stopped` state to service manager on windows.

## Checklist


## How to test
Install agent as a service on windows and try following scenarios
- restart windows machine 
- restart agent from service management
- restart agent using `./elastic-agent restart`
- kill agent from Task manager

Check event viewer, `The Elastic Agent service terminated unexpectedly` should appear only for last scenario as this is the only time agent was not expected to shut down.

![image](https://user-images.githubusercontent.com/1522652/229155569-4cb3f0d7-cdde-4e50-94e4-5c11024951e6.png)

Fixes: https://github.com/elastic/elastic-agent/issues/1525
